cmd/operator-ui/vendor/github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/v1/containers/imageTags.go

```diff
@@ -0,0 +1,70 @@
+package containers
+
+import "fmt"
+
+var imageTags = map[string]map[string]string{
+	"2019.2.0": {
+		"blackduck-authentication": "2019.2.0",
+		"blackduck-documentation":  "2019.2.0",
+		"blackduck-jobrunner":      "2019.2.0",
+		"blackduck-registration":   "2019.2.0",
+		"blackduck-scan":           "2019.2.0",
+		"blackduck-webapp":         "2019.2.0",
+		"blackduck-cfssl":          "1.0.0",
+		"blackduck-logstash":       "1.0.2",
+		"blackduck-nginx":          "1.0.2",
+		"blackduck-solr":           "1.0.0",
+		"blackduck-zookeeper":      "1.0.0",
+		"blackduck-upload-cache":   "1.0.3",
+		"appcheck-worker":          "1.0.1",
+		"rabbitmq":                 "1.0.0",
+	},
+	"2019.2.1": {
+		"blackduck-authentication": "2019.2.1",
+		"blackduck-documentation":  "2019.2.1",
+		"blackduck-jobrunner":      "2019.2.1",
+		"blackduck-registration":   "2019.2.1",
+		"blackduck-scan":           "2019.2.1",
+		"blackduck-webapp":         "2019.2.1",
+		"blackduck-cfssl":          "1.0.0",
+		"blackduck-logstash":       "1.0.2",
+		"blackduck-nginx":          "1.0.2",
+		"blackduck-solr":           "1.0.0",
+		"blackduck-zookeeper":      "1.0.0",
+		"blackduck-upload-cache":   "1.0.3",
+		"appcheck-worker":          "1.0.1",
+		"rabbitmq":                 "1.0.0",
+	},
+	"2019.2.2": {
+		"blackduck-authentication": "2019.2.2",
+		"blackduck-documentation":  "2019.2.2",
+		"blackduck-jobrunner":      "2019.2.2",
+		"blackduck-registration":   "2019.2.2",
+		"blackduck-scan":           "2019.2.2",
+		"blackduck-webapp":         "2019.2.2",
+		"blackduck-cfssl":          "1.0.0",
+		"blackduck-logstash":       "1.0.2",
+		"blackduck-nginx":          "1.0.2",
+		"blackduck-solr":           "1.0.0",
+		"blackduck-zookeeper":      "1.0.0",
+		"blackduck-upload-cache":   "1.0.3",
+		"appcheck-worker":          "1.0.1",
+		"rabbitmq":                 "1.0.0",
+	},
+}
+
+func (c *Creater) getImageTag(name string) string {
+	confImageTag := c.GetFullContainerNameFromImageRegistryConf(name)
+	if len(confImageTag) > 0 {
+		return confImageTag
+	}
+	return fmt.Sprintf("docker.io/blackducksoftware/%s:%s", name, imageTags[c.hubSpec.Version][name])
```

Comment:
Why the docker.io/blackducksoftware is hard coded? The upstream hub testing will use images from gcr.io/saas-hub-stage/blackducksoftware and also there are customers who don't have access to docker hub in their cluster, so they try to push to their internal docker registry and use those hub images.

Reply:
This hasn't changed and still works the same way. We default to using docker.io and if a customer wants to use a different registry then they need to specify the images in the  "imageRegistries" config.

Reply:
Got it..